blog/build-a-network-speed-monitor-in-bash-with-awk.md

```diff
@@ -0,0 +1,110 @@
+# Build a network speed monitor in bash with AWK
+
+Sometimes we need to measure the network speed of a server but some services like `speedtest` are not useful because we want to check our bandwidth with another cloud service and not the internet itself.
+
+## AWK to the rescue
+
+We can write a simple script in AWK to parse the `/proc/net/dev` pseudo-file which contains network device status information. AWK is a [programming language](https://dev.to/rrampage/awk---a-useful-little-language-2fhf) for manipulating columns of data that we can't solve with a simple `sed` command and a regex.
+
+But before talk about the AWK script, we should define the sentences in bash that we need in this script. Or, you can jump to the end to find the complete script 😁.
+
+## The bash part
+
+We need to create a `.sh` file with an infinite loop to print the network speed every 2 seconds.
+
+```bash
+INTERFACE=eth0
+echo "$INTERFACE  down (KiB/s)   up (KiB/s)"
+while :
+do
+  # The AWK command here!
+  sleep 2;
+done
+```
+
+First, we save the name of network interface to inspect in a `INTERFACE` variable, in this example is `eth0`.
+
+Second, we print a nice header to remember us which column is the download speed (and the upload one) and the speeds are measured in kilobytes per second (KiB/s, the `i` is because [1 Kibibyte is 1024 bytes](https://en.wikipedia.org/wiki/Byte#Units_based_on_powers_of_2)).
```

Comment:
```suggestion
Then, we print a nice header to remind ourselves which column is the download speed (and the upload one). These speeds are measured in kilobytes per second (KiB/s, the `i` is because [1 Kibibyte is 1024 bytes](https://en.wikipedia.org/wiki/Byte#Units_based_on_powers_of_2)).
```